src/main/java/uk/gov/gchq/palisade/util/AbstractResourceBuilder.java

```diff
@@ -27,11 +27,12 @@
 import java.util.ServiceLoader.Provider;
 
 /**
- * ResourceBuilder, taking a URI and building a Resource specific to each scheme
+ * The AbstractResourceBuilder is an abstract factory for building {@link Resource}s. Given a URI,
+ * it will identify the appropriate implementation and use it to produce a {@link Resource}
```

Comment:
I'd add a note here about it using java's ServiceLoader mechanism
https://docs.oracle.com/javase/7/docs/api/java/util/ServiceLoader.html

Reply:
Done